core/ipc/rpc/utils/src/main/java/org/opennms/core/rpc/utils/mate/Interpolator.java

```diff
@@ -52,10 +54,18 @@ private Interpolator() {}
     }
 
     public static Object interpolate(final Object value, final Scope scope) {
-        if (value instanceof String) {
+        if (value == null) {
+            return null;
+        } else if (value instanceof String) {
             return interpolate((String) value, scope);
         } else {
-            return value;
+            // Serialize, interpolate and deserialize the value
+            final Class<?> clazz = value.getClass();
+            final String xml = JaxbUtils.marshal(value);
```

Comment:
Should we try catching any exceptions that may happen during this process and return the original value if something fails, or leave that up to the caller?

(I suspect the marshaling/unmarshaling may break with some types of objects or in certain circumstances.)

Reply:
God idea. I'm not sure which exceptions to catch? But `Throwable` should do the trick, right?